src/remarx/get_token_embeddings.py

```diff
@@ -0,0 +1,301 @@
+"""
+Script to build embeddings of a specific term for each sentence in a
+sentence-level corpus file (JSONL). The resulting embeddings are saves as a .npy
+binary file. By default the  MT5-XL model (google/mt5-xl) is used, but this can
+be changed to another pretrained model. Since term instances and sentences are
+unlikely to have a one-to-one correspondence, a metadata file (CSV) identifying
+each embedding is also generated.
+
+See https://huggingface.co/docs/transformers/main/en/model_doc/mt5 for more on MT5.
+
+Example:
+
+    python get_token_embeddings sents.jsonl Kapital output_dir output_stem
+
+"""
+
+import argparse
+import csv
+import pathlib
+import re
+import sys
+from typing import cast
+
+import numpy as np
+import numpy.typing as npt
+import orjsonl
+from tqdm import tqdm
+from transformers import (
+    FeatureExtractionPipeline,
+    PreTrainedTokenizer,
+    T5Tokenizer,
+    pipeline,
+)
+
+
+def normalize_text(text: str) -> str:
+    """
+    Normalizes text so that it will align with its tokenized version.
+    """
+    # Remove trailing whitespace
+    result_text = text.rstrip()
+    # Replace consecutive whitespace with a space
+    result_text = " ".join(result_text.split())
+    # Add leading space
+    result_text = " " + result_text
+    return result_text
```

Comment:
one-line version, fwiw. I hadn't thought about adding more steps later, but for this set of steps it seems pretty readable.

```suggestion
    # Remove any outer whitespace, consolidate multiple whitespace characters, and add a leading space
     return " " + " ".join(text.strip().split())
```

Reply:
You've convinced me. Thanks.